src/reducers/AddSiteReducer.js

```diff
@@ -10,44 +12,63 @@ import {
     SITE_ACCESSIBILITY_OPTION_CHANGED,
     SITE_FACILITIES_OPTION_CHANGED,
     ADD_SITE_FIELDS_RESET,
-    CURRENT_LOCATION_UPDATED
+    CURRENT_LOCATION_UPDATED,
+    CHECK_IF_SITE_IS_READY,
+    ADD_SITE_SUCCESS,
+    ADD_SITE_FAILURE
 } from '../actions/types';
 
-import {campsite} from '../locale.en';
+import {campsite, reducerAlerts} from '../locale.en';
 
 const {campsite_form: {accessibility_options, facilities_options}} = campsite;
 
 const INITIAL_STATE = {
+    addSiteModalVisible: false,
     latitudeText: '',
+    readyLatitude: 0,
     longitudeText: '',
+    readyLongitude: 0,
     siteTitleText: '',
     siteDescriptionText: '',
     siteDirectionsText: '',
     siteNearestTownText: '',
     accessibilityOption: accessibility_options.paved_road,
-    facilitiesOption: facilities_options.full_service
+    facilitiesOption: facilities_options.full_service,
+    siteReadyForUpload: false
 };
 
 const removeNonNumbers = (text) => {
     return text.replace(/[^0-9.-]/g, '');
 };
 
+const siteIsReadyForUpload = ({latitudeText, longitudeText, siteTitleText, siteDescriptionText, siteDirectionsText, siteNearestTownText, accessibilityOption, facilitiesOption}) => {
+    return latitudeText && longitudeText && siteTitleText && siteDescriptionText && siteDirectionsText && siteNearestTownText && accessibilityOption && facilitiesOption;
+};
+
 export default (state = INITIAL_STATE, action) => {
     const {type, payload} = action;
 
     switch (type) {
 
+        case OPEN_SITE_UPLOAD_MODAL:
+            return {...state, addSiteModalVisible: true};
+
+        case CLOSE_SITE_UPLOAD_MODAL:
+            return {...state, addSiteModalVisible: false};
+
         case LATITUDE_TEXT_UPDATED:
             const {latitudeText} = payload;
             const cleanLatText = removeNonNumbers(latitudeText);
+            const readyLatitude = _.toNumber(cleanLatText);
 
-            return {...state, latitudeText: cleanLatText};
+            return {...state, latitudeText: cleanLatText, readyLatitude};
 
         case LONGITUDE_TEXT_UPDATED:
             const {longitudeText} = payload;
             const cleanLongText = removeNonNumbers(longitudeText);
+            const readyLongitude = _.toNumber(cleanLongText);
 
-            return {...state, longitudeText: cleanLongText};
+            return {...state, longitudeText: cleanLongText, readyLongitude};
```

Comment:
`readyLatitude` and `readyLongitude` are the `Number` representations of whatever the user entered. This is what we're actually going to pass along with the object instead of the text. Let's update it every time the lat/long text is updated.

Reply:
nifty!